communicator/ssh/communicator.go

```diff
@@ -344,10 +364,69 @@ func (c *comm) reconnect() (err error) {
 		c.client = ssh.NewClient(sshConn, sshChan, req)
 	}
 	c.connectToAgent()
+	c.connectTunnels(sshConn)
 
 	return
 }
 
+func (c *comm) connectTunnels(sshConn ssh.Conn) {
+	if c.client == nil {
+		return
+	}
+
+	// Start remote forwards of ports to ourselves.
+	log.Printf("[DEBUG] Tunnel Configuration: %v", c.config.Tunnels)
+	for _, v := range c.config.Tunnels {
+		done := make(chan struct{})
+		switch v.Direction {
+		case RemoteTunnel:
+			// This requests the sshd Host to bind a port and send traffic back to us
+			listener, err := c.client.Listen(v.ListenType, v.ListenAddr)
+			// TODO How can we get this failure to ui.Error?
```

Comment:
Answer: you can change the function signature of connectTunnels to return an error. you're calling connectTunnels inside reconnect(), which already returns an error, so with a minor tweak to the connectTunnels() callling code in reconnect() you can return any errors surfaced. I'm pretty sure that in a majority of cases, errors returned by reconnect() will already get surfaced through the UI, bubbling up through the calls like Download() or whatever and getting handled within the plugins that use the communicator.